azurelinuxagent/ga/update.py

```diff
@@ -1016,8 +1017,11 @@ def _check_agent_memory_usage(self):
         """
         try:
             if conf.get_enable_agent_memory_usage_check() and self._extensions_summary.converged:
-                if self._last_check_memory_usage == datetime.min or datetime.utcnow() >= (self._last_check_memory_usage + UpdateHandler.CHECK_MEMORY_USAGE_PERIOD):
-                    self._last_check_memory_usage = datetime.utcnow()
+                # we delay first attempt memory usage check, so that current agent won't get blacklisted due to multiple restarts(because of memory limit reach) too frequently
+                if (self._initial_attempt_check_memory_usage and time.time() - self._last_check_memory_usage_time > CHILD_LAUNCH_INTERVAL) or \
```

Comment:
why the change from utcnow() to time()? - the former has the advantage that it formats nicely when, for example, running the code under the debugger

Reply:
Nothing specific, I just came across the idea. Let me update if it has advantage

Reply:
ok, it is not a big issue.